docs: Ignore auto-generated .rst files for python lib docs

echoix opened the pull request:
When building the python library docs locally with `make sphinxdoclib`, some .rst files are generated and not ignored.

Ending up with files to commit like this:
![image](https://github.com/user-attachments/assets/3e00cedf-012f-4f6a-b205-b4c8c16704a9)

For the last two weeks, I've used this addition to the .gitignore. Each file was individually added, as there are some other files committed that aren't generated by autodocs/autoapi (or any other tool that is part of sphinx).

It worked well enough that I think it is worth it. I get reminded how these aren't excluded when switching back to the main branch to create an independent branch

cwhite911 reviewed (commented):
Check what is getting cleaned by 
_libpythonclean_. We may be able to shorten the number of entries. I'm looking at this from my phone on a plane otherwise I would do it.

echoix commented:
Can I get a new look after addressing the comments two weeks ago?
Now, it ignores the same things that is deleted at each compiling, using the same patterns as in the makefiles (adapted). So there's nothing changed really.
